src/sidebar/components/MarkdownView.tsx

```diff
@@ -22,14 +44,49 @@ export default function MarkdownView({
   markdown,
   classes,
   style,
-  mentions,
+  mentions = [],
+  mentionsEnabled = false,
+  /* istanbul ignore next - test seam */
+  setTimeout_ = setTimeout,
+  /* istanbul ignore next - test seam */
+  clearTimeout_ = clearTimeout,
 }: MarkdownViewProps) {
   const html = useMemo(
     () => (markdown ? renderMathAndMarkdown(markdown) : ''),
     [markdown],
   );
   const content = useRef<HTMLDivElement | null>(null);
 
+  const mentionsPopoverAnchorRef = useRef<HTMLElement | null>(null);
+  const elementToMentionMap = useRef(
+    new Map<HTMLElement, Mention | InvalidUsername>(),
+  );
+  const [popoverContent, doSetPopoverContent] = useState<PopoverContent>(null);
+  const popoverContentTimeout = useRef<ReturnType<typeof setTimeout> | null>();
+  const setPopoverContent = useCallback(
```

Comment:
There is nothing to clear the timer if the component gets unmounted after a hover has started but before the popover has been shown. You could use a layout effect cleanup function to handle this. A completely different approach would be to use an animation on the popover container to cause the delayed appearance (ie. the 400ms delay would become a transition on eg. `visibility`).

Reply:
I'll add the effect cleanup function and check how difficult would be to use a native animation afterwards.